pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/TableConfigUtils.java

```diff
@@ -580,4 +616,124 @@ private static IndexingConfig sanitizeIndexingConfig(IndexingConfig indexingConf
+        if (replicasPerPartition < defaultTableMinReplicas) {
+          LOGGER.info(
+              "Creating table with minimum replicasPerPartition of: {} instead of requested replicasPerPartition: {}",
+              defaultTableMinReplicas, replicasPerPartition);
+          segmentsConfig.setReplicasPerPartition(String.valueOf(defaultTableMinReplicas));
+        }
+      } catch (NumberFormatException e) {
+        throw new IllegalStateException("Invalid value for replicasPerPartition: '" + replicasPerPartitionStr + "'", e);
+      }
+    }
+  }
+
+  /**
+   * Ensure the table config has storage quota set as per cluster configs.
+   * If it doesn't, set the quota config into the table config
+   */
+  public static void ensureStorageQuotaConstraints(TableConfig tableConfig, String maxAllowedSize) {
+    // Dim tables must adhere to cluster level storage size limits
+    if (tableConfig.isDimTable()) {
+      QuotaConfig quotaConfig = tableConfig.getQuotaConfig();
+      long maxAllowedSizeInBytes = DataSizeUtils.toBytes(maxAllowedSize);
+
+      if (quotaConfig == null) {
+        // set a default storage quota
+        tableConfig.setQuotaConfig(new QuotaConfig(maxAllowedSize, null));
+        LOGGER.info("Assigning default storage quota ({}) for dimension table: {}", maxAllowedSize,
+            tableConfig.getTableName());
+      } else {
+        if (quotaConfig.getStorage() == null) {
+          // set a default storage quota and keep the RPS value
+          tableConfig.setQuotaConfig(new QuotaConfig(maxAllowedSize, quotaConfig.getMaxQueriesPerSecond()));
+          LOGGER.info("Assigning default storage quota ({}) for dimension table: {}", maxAllowedSize,
+              tableConfig.getTableName());
+        } else {
+          if (quotaConfig.getStorageInBytes() > maxAllowedSizeInBytes) {
+            throw new IllegalStateException(String
+                .format("Invalid storage quota: %d, max allowed size: %d", quotaConfig.getStorageInBytes(),
+                    maxAllowedSizeInBytes));
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Consistency checks across the offline and realtime counterparts of a hybrid table
+   */
+  public static void verifyHybridTableConfigs(String rawTableName, TableConfig offlineTableConfig,
+      TableConfig realtimeTableConfig) {
+    if (offlineTableConfig == null || realtimeTableConfig == null) {
+      return;
+    }
+
+    LOGGER.info("Validating realtime and offline configs for the hybrid table: {}", rawTableName);
+    String offlineRawTableName = TableNameBuilder.extractRawTableName(offlineTableConfig.getTableName());
+    String realtimeRawTableName = TableNameBuilder.extractRawTableName(realtimeTableConfig.getTableName());
+    Preconditions.checkState(offlineRawTableName.equals(realtimeRawTableName),
+        "Raw table name for offline table: %s does not match raw table name for realtime table: %s");
```

Comment:
These validation are already performed on the caller side. We can only validate the cross table part (time column match)